lib/puppet/application/lookup.rb

```diff
@@ -33,9 +33,12 @@ class Puppet::Application::Lookup < Puppet::Application
 
   option('--merge_hash_arrays')
 
-  # not yet supported
   option('--explain')
 
+  option('--explain_json')
```

Comment:
The only command that uses `--format` is AFAICS the 'doc' command and there it seems to have a slightly different meaning (renders pdf or text). A more commonly used pattern for what we want to achieve here is to use `--render-as <format>` and then map that to a renderer returned by `Puppet::Network::FormatHandler.format()`. The user then has the option to use _s_ (text), _json_, _yaml_, _msgpack_ and _binary_ per default.